File: portlets/sample-python-portlet/docroot/WEB-INF/python/info.py
renderResponse.setContentType("text/html");

out = renderResponse.getPortletOutputStream();

params = renderRequest.getParameterMap();

custom = Custom(renderRequest, renderResponse);
lp = LiferayPortlet(renderRequest, renderResponse);

out.write(custom.sayHello() + "\n");

out.write("<br />");

if (params['type'] and params['type'][0] == 'user'):
	if (userInfo != None):
	   out.write(lp.showUserDetails(userInfo) + "\n");
	else:
	   out.write("You are not signed in.<br />");
else:
	out.write(lp.showPortletDetails(portletConfig) + "\n");

out.write(custom.navigation() + "\n");

File: portlets/sample-javascript-portlet/docroot/WEB-INF/javascript/info.js
renderResponse.setContentType("text/html");

out = renderResponse.getPortletOutputStream();

out.println(Custom.sayHello());

out.println("<br />");

params = renderRequest.getParameterMap();

if (params.get('type') && (params.get('type')[0] == 'user')) {
	if (userInfo != null) {
		out.println(LiferayPortlet.showUserDetails(userInfo));
	} else {
		out.println("You are not signed in<br />");
	}
}
else {
	out.println(LiferayPortlet.showPortletDetails(portletConfig, renderRequest));
}

out.println(Custom.navigation(renderResponse));